src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -392,38 +363,7 @@ private TElement Last(Buffer<TElement> buffer)
             return value;
         }
 
-        public TElement Last(Func<TElement, bool> predicate)
-        {
-            CachingComparer<TElement> comparer = GetComparer();
-            using (IEnumerator<TElement> e = _source.GetEnumerator())
-            {
-                TElement value;
-                do
-                {
-                    if (!e.MoveNext())
-                    {
-                        throw Error.NoMatch();
-                    }
-
-                    value = e.Current;
-                }
-                while (!predicate(value));
-
-                comparer.SetElement(value);
-                while (e.MoveNext())
-                {
-                    TElement x = e.Current;
-                    if (predicate(x) && comparer.Compare(x, false) >= 0)
```

Comment:
It should be `>=`, (while `First` uses `<` as they need to have opposite behaviour on tie-breaking to match the behaviour of doing a stable sort followed by getting the last or first item respectively).